lib/sla.js

```diff
@@ -45,80 +48,135 @@ function designateMixingBridge(client, bridgeName) {
+    object.channel.on('ChannelHold', channelHold);
     console.log('Channel %s has entered the bridge', object.channel.id);
-    numInBridge += 1;
-  }
 
-  bridge.on('ChannelEnteredBridge', channelEntered);
+  }
 
+  /**
+   * Function that is utilized when a channel leaves the bridge.
+   *   If the channel is a non-station channel, then hangs up remaining
+   *   channels.  If the channel is a station, checks if the only non-station
+   *   channel is the only left, and then hangs up that non-station channel.
+   * @param {Object} event - the event related to the bridge leaving
+   * @param {Object} object - contains the bridge and channel in question
+   */
   function channelLeft(event, object) {
-    numInBridge -= 1;
-    console.log('Channel %s has left the bridge', object.channel.id);
-
-    if (numInBridge === 1) {
-      if (object.channel.id === inbound.id) {
-        console.log('Hanging up dialed channel %s', dialed.id);
-        var hangupDialed = Q.denodeify(dialed.hangup.bind(dialed));
-        hangupDialed()
-          .catch(function (err) {
-            err.name = 'HangupFailure';
-            defer.reject(err);
-          });
-      }
-      else if (object.channel.id === dialed.id) {
-        console.log('Hanging up inbound channel %s', inbound.id);
-        var hangupInbound = Q.denodeify(inbound.hangup.bind(inbound));
-        hangupInbound()
-          .catch(function (err) {
-            err.name = 'HangupFailure';
-            defer.reject(err);
+    object.channel.removeAllListeners('ChannelHold');
+
+    isStation(extension, object.channel).then(function (isStation) {
+      if (isStation) {
+        if (object.bridge.channels.length === 1) {
+          object.bridge.startMoh();
+          updateState(client, bridge.name, hold);
+        }
+
+      } else {
+        if (object.bridge.channels.length) {
+          console.log('Hanging up station channel/s');
+
+          object.bridge.channels.forEach( function(ID) {
+
+            var hangup = Q.denodeify(client.channels.hangup.bind
+              (client.channels));
+            hangup({channelId: ID})
+              .catch(function (err) {
+                err.name = 'HangupFailure';
+                defer.reject(err);
+              });
           });
+        }
       }
-    }
 
-    if (numInBridge === 0) {
-      updateState(client, bridge.name, idle);
-      defer.resolve('Application completed');
-      object.bridge.removeListener('ChannelEnteredBridge', channelEntered);
-      object.bridge.removeListener('ChannelLeftBridge', channelLeft);
-    }
+      if (object.bridge.channels.length === 0) {
+        defer.resolve(true);
+        updateState(client, bridge.name, idle);
+        bridge.removeAllListeners('ChannelEnteredBridge');
+        bridge.removeAllListeners('ChannelLeftBridge');
+      }
+    });
+  }
+  function channelHold(event, channel) {
+    isStation(extension, channel).then(function (isStation) {
+      if (isStation) {
```

Comment:
You shouldn't have to check if this is a station or not, since you control whose holds you intercept based on who you put the function on.